packages/flame/lib/src/experimental/geometry/shapes/polygon.dart

```diff
@@ -241,4 +243,35 @@ class Polygon extends Shape {
 
   @override
   String toString() => 'Polygon($vertices)';
+
+  @override
+  Vector2 randomPoint({Random? random, bool within = true}) {
+    final randomGenerator = random ?? randomFallback;
+    if (within) {
+      final result = Vector2.zero();
+      final min = aabb.min;
+      final max = aabb.max;
+
+      while (true) {
+        final randomX = min.x + randomGenerator.nextDouble() * (max.x - min.x);
+        final randomY = min.y + randomGenerator.nextDouble() * (max.y - min.y);
+        result.setValues(randomX, randomY);
+
+        if (containsPoint(result)) {
+          return result;
+        }
+      }
+    } else {
+      final edgeIndex = randomGenerator.nextInt(vertices.length);
```

Comment:
not uniformly distributed bc the edges can have different lengths and thus should have different weights.
how about adding up the perimeter, generate a random number and find the edge and then find the point

Reply:
True true! That will be quite expensive though, I guess if we use length2 it shouldn't be a problem.